src/main/java/org/broadinstitute/hellbender/utils/read/CigarUtils.java

```diff
@@ -430,4 +430,20 @@ public static int countAlignedBases(final Cigar cigar ) {
                 .mapToInt(CigarElement::getLength)
                 .sum();
     }
+
+    /**
+     * replace soft clips (S) with match (M) operators, merging any consecutive M's that result eg 10S10M -> 20M.
+     */
+    public static Cigar revertSoftClips(final Cigar originalCigar) {
```

Comment:
> This should note that it will also normalize the cigar in other ways?

done

> This will explode on a read that is entirely soft clips. I don't know if that important.

Completely clipped reads are treated as invalid elsewhere in the code.  I think I agree with this because if you have no aligned base the read should be considered unmapped.  But also, a completely soft-clipped read will send a bunch of `M` elements to the `CigarBuilder` and not explode, I think.  Actually, you could complain about it failing to throw an error if the input cigar is completely clipped, but it don't think this method should be responsible for that.